src/js/navButtons.js

```diff
@@ -163,4 +163,34 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         that.applier.change("currentPanelNum", newValue);
     };
 
+    // A grade component for integrating the first discovery tool and the preferences server.
+    // It overrides some definitions on "gpii.firstDiscovery.navButtons"
+    fluid.defaults("gpii.firstDiscovery.navButtons.prefsServerIntegration", {
+        gradeNames: ["fluid.modelComponent"],
+        invokers: {
+            indexToDisposition: {
+                funcName: "gpii.firstDiscovery.navButtons.prefsServerIntegration.indexToDisposition",
+                args: ["{that}.model.currentPanelNum", "{that}.options.panelStartNum", "{that}.options.panelTotalNum"]
+            }
+        },
+        modelListeners: {
+            isLastPanel: [{
+                listener: "{that}.toggleButtonStates",
+                args: ["{that}.dom.next", false],
+                excludeSource: "init",
+                priority: 3
+            }]
+        },
+        listeners: {
+            "onCreate.toggleNextButtonStates": {
+                listener: "{that}.toggleButtonStates",
+                args: ["{that}.dom.next", false]
+            }
+        }
+    });
+
+    gpii.firstDiscovery.navButtons.prefsServerIntegration.indexToDisposition = function (currentPanelNum, panelStartNum, panelTotalNum) {
+        return currentPanelNum === panelStartNum ? 0 : (currentPanelNum < panelTotalNum ? 1 : 2);
+    };
```

Comment:
Could you leave a comment about what these numbers "0, 1, 2" are for. We may actually want to setup "constants" for them.